tests/test_structured_state_conditionals.py

```diff
@@ -26,6 +27,36 @@
 # Further down, there are tests to check that the simulation works correctly.
 
 
+def test_circuit_with_clexpr_i() -> None:
+    # test conditional handling
+
+    circ = Circuit(3)
+    a = circ.add_c_register("a", 5)
+    b = circ.add_c_register("b", 5)
+    c = circ.add_c_register("c", 5)
+    d = circ.add_c_register("d", 5)
+    circ.H(0)
+    wexpr, args = wired_clexpr_from_logic_exp(a | b, c)  # type: ignore
```

Comment:
I had to add `# type: ignore` here because it complains that the second argument `c` is a `BitRegister`, rather than `list[Bit]`. Is there a way around it? Is this a valid implicit cast, or should it be avoided?

Reply:
You could write `c.to_list()`.